### PR TITLE
net: JoinHostPort() and MakeHostPort()

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: splithostport.go [hunk 1]
```diff
@@ -10,6 +10,64 @@ import (
...
+		// bad input
+		return "", err
+	}
+
+	if ip, _ := ParseAddr(host); ip.IsValid() {
+		// ip address
+		return doMakeAddrPort(ip, port, defaultPort)
+	}
+
+	s, ok := validName(host)
```
Comment (amery) [inline splithostport.go:28]: this is redundant as `SplitHostPort` above already validated it

**Comment by amery** [comment]
simplified.

**Comment by amery** [comment]
due to possible use cases `JoinHostPort` takes "0" as a valid port, because there is the intent of the developer to always have a port in the string, while `foo:0` being invalid for `MakeHostPort()` because that probably comes from configuration

**Review by karasz — APPROVED** [review]
LGTM

**Comment by amery** [comment]
`v0.14.3` tagged

**Pull request merged**